esp-hal/src/dma/buffers.rs

```diff
@@ -14,16 +14,16 @@ pub struct Preparation {
     #[cfg_attr(not(esp32s3), allow(dead_code))]
     pub block_size: Option<DmaBufBlkSize>,
 
-    /// Specifies whether descriptor linked list specified in `start` conforms
-    /// to the alignment requirements required to enable burst transfers.
+    /// Specifies whether the data should be transferred in burst mode.
     ///
-    /// Note: This only applies to burst transfer of the buffer data, not the
-    /// descriptors themselves.
+    /// The implementation of the buffer must ensure that burst mode is only
+    /// enabled when alignment requirements are met.
     ///
     /// There are no additional alignment requirements for TX burst transfers,
     /// but RX transfers require all descriptors to have buffer pointers and
     /// sizes that are a multiple of 4 (word aligned).
-    pub is_burstable: bool,
+    // TODO: currently unused, buffers should not hardcode their preference.
+    pub burst: bool,
```

Comment:
The way the alignment requirements are, and how only psram has a block size on the S3, this is extremely annoying. We could ideally describe the whole burst mode configuration as an `Option<BlockSize>`, but the valid values depend on a bunch of factors (like, for SRAM, this is a boolean, for PSRAM it's forced, on P4 it depends on the DMA used) and I'm not sure what to do here.

I'm considering turning this into a "you are allowed to burst, and pick a size at most this" configuration - that way we have an `Option<BlockSize>`, and the hardware may do its own thing if it is not happy. We can return an error if the user wants burst transfers but the hardware absolutely can't do it.

Reply:
I guess we can figure this out later, too.

Reply:
Don't forget you can also mix SRAM and PSRAM in the same DMA linked list, so you still need two separate configs

Reply:
😭